core/conversion/converters/impl/plugins/interpolate_plugin.cpp

```diff
@@ -0,0 +1,244 @@
+   nvinfer1::DimsExprs output(inputs[0]);
+
+   for (unsigned int i = 0; i < out_shape.size(); i++) {
+       output.d[i] = exprBuilder.constant(out_shape[i]);
+   }
+
+   return output;
+}
+
+nvinfer1::DataType InterpolatePlugin::getOutputDataType(int index, const nvinfer1::DataType* inputTypes, int nbInputs) const {
+    return DataType::kFLOAT;
+}
+
+int InterpolatePlugin::initialize() {
+    tensor_options = tensor_options.device(c10::kCUDA);
+
+    // c10::kFloat = FLOAT32
+    tensor_options = tensor_options.dtype(c10::kFloat);
+
+    return 0;
+}
+
+
+void InterpolatePlugin::serialize(void* buffer) const {
+    std::string data = serializeToString();
+    size_t size = getSerializationSize();
+
+    data.copy((char*) buffer, size);
+}
+
+std::string InterpolatePlugin::serializeToString() const {
+    torch::serialize::OutputArchive output_archive;
+
+    output_archive.write("in_shape", torch::IValue(in_shape));
+    output_archive.write("out_shape", torch::IValue(out_shape));
+    output_archive.write("size", torch::IValue(size));
+    output_archive.write("mode", torch::IValue(mode));
+    output_archive.write("align_corners", torch::IValue(align_corners));
+
+    std::ostringstream data_str;
+    output_archive.save_to(data_str);
+
+    return data_str.str();
+}
+
+size_t InterpolatePlugin::getSerializationSize() const {
+    return serializeToString().size();
+}
+
+bool InterpolatePlugin::supportsFormatCombination(int pos, const nvinfer1::PluginTensorDesc* inOut, int nbInputs, int nbOutputs) {
+    if (inOut->format != nvinfer1::TensorFormat::kLINEAR) {
+        return false;
+    } 
+
+    if (inOut->type == DataType::kINT32 || inOut->type == DataType::kINT8) {
+        return false;
+    }
+
+    return true;
+}
+
+void InterpolatePlugin::configurePlugin(const nvinfer1::DynamicPluginTensorDesc* in, int nbInputs, const nvinfer1::DynamicPluginTensorDesc* out, int nbOutputs) {
+    dtype = DataType::kFLOAT;
+}
+
+size_t InterpolatePlugin::getWorkspaceSize(const nvinfer1::PluginTensorDesc* inputs, int nbInputs, const nvinfer1::PluginTensorDesc* outputs, int nbOutputs) const {
+    return 0;
+}
+
+int InterpolatePlugin::enqueue(const nvinfer1::PluginTensorDesc* inputDesc, const nvinfer1::PluginTensorDesc* outputDesc, const void *const *inputs, 
+                                                                                                        void *const *outputs, void *workspace, 
+                                                                                                        cudaStream_t stream) {
+    at::Tensor input;
+
+    if (mode == "adaptive_pool2d") {
+        // use dynamically inferred input shape (for pooling)
+        input = at::from_blob((void*) inputs[0], util::toVec(inputDesc->dims), [](void*){}, tensor_options);
+    } else {
+        // use precomputed input shape (for interpolation/upsampling)
+        input = at::from_blob((void*) inputs[0], in_shape, [](void*){}, tensor_options);

```

Comment:
Is there a reason why we couldn't use inputDesc for both cases?

Reply:
was thinking about this - yeah, can do for both. this would mean interpolate would also support dynamically sized inputs